### ISSUE TITLE
Possible to make whole page activate as jupyter lite

### ISSUE THREAD

**Issue by ianhi** [martinRenou/jupyterlite-sphinx#2]
Is it's possible to make the entire page active as a jupyter kernel? So if I wrote the docs as a notebook then they just become active.

Currently the jupyter part is contained: 
![image](https://user-images.githubusercontent.com/10111092/155740335-14819d3a-bf04-4358-aa5b-e8d6f02f7052.png)

From this thebe issue: https://github.com/executablebooks/thebe/issues/465#issuecomment-908079835 I guess the struggle is that this would require implementing a new frontend to communicate with the backend?

Or maybe a wilder idea is a jlab extension to make jupyterlab look like a docs website, then it would be easy to have jupyterlite run everything.

**Comment by martinRenou** [comment]
Indeed, it would be nice to share one kernel for all the consoles in the page 👍🏽

**Comment by bollwyvl** [comment]
> look like a docs website, 

right, i think that's achievable, but probably out of scope for this initial go-around. the prototype form is https://github.com/jupyterlite/jupyterlite/issues/148, but we imagine extending it to a JSON schema for composing custom jupyter experiences from top-level applications. 

>  share one kernel for all the consoles

this challenge is being discussed here: https://github.com/jupyterlite/jupyterlite/issues/463. the idea would be to offer more than just "iframe kernel," _or_ "webworker kernel," but a composite of "webworker-in-an-iframe kernel" as a stand-in for "sharedworker kernel," which is not widely-distributed yet due to (sound) limitations by the browser security model.

**Comment by amueller** [comment]
So is there a downside to do the whole thing within a jupyter lab and have a plugin that renders the docs similar to what is shown in sphinx?
